src/io.cc

```diff
@@ -161,3 +161,43 @@ std::string mpm::IO::output_folder() const {
   }
   return path;
 }
+
+//! Return map of entity sets from the JSON file
+std::map<Index, std::vector<Index>> mpm::IO::entity_sets(
+    const std::string& filename, const std::string& sets_type) {
+
+  // Input file stream for sets JSON file
+  std::ifstream sfs;
+  sfs.exceptions(std::ifstream::failbit | std::ifstream::badbit);
+  // Map of the entity sets
+  std::map<Index, std::vector<Index>> entity_sets;
+
+  try {
+    sfs.open(filename);
+    // Entity sets JSON object
+    Json json_sets_ = Json::parse(sfs);
+    // type of sets (ex: node sets or particle sets)
+    Json esets = json_sets_["sets_type"];
+
+    if (esets.size() > 0) {
+      for (Json::iterator itr = esets.begin(); itr != esets.end(); ++itr) {
+        // Get the entity set ids
+        Index id = (*itr)["id"].template get<Index>();
+        // Get the entity ids
+        std::vector<Index> eids;
+        // Get the size of the set vector
+        eids.resize((*itr).at("set").size());
+
+        for (unsigned i = 0; i < (*itr).at("set").size(); ++i)
+          eids.emplace_back((*itr).at("set").at(i).template get<Index>());
+
+        entity_sets[id] = eids;
```

Comment:
I thin map has an insert function, which returns a boolean, could we try to use that?

Reply:
http://www.cplusplus.com/reference/map/map/insert/

Reply:
Ok, I will